localstack/config.py

```diff
@@ -32,6 +32,115 @@
+        tmp:        individual; ephemeral data that does not have to persist across localstack runs but not reboots
+        shared_tmp: shared; ephemeral data used to communicate between container<->lambdas
+        data:       shared; holds localstack state, pods, ...
+        config:     host only; pre-defined configuration values, cached credentials, machine id, ...
+        init:       shared; user-defined provisioning scripts executed when in the container when it starts
+        logs:       shared; log files produced by localstack
+    """
+
+    infra: str
+    var: str
+    cache: str
+    tmp: str
+    shared_tmp: str
+    data: str
+    config: str
+    init: str
+    logs: str
+
+    def __init__(
+        self,
+        infra: str = None,
+        var: str = None,
+        cache: str = None,
+        tmp: str = None,
+        shared_tmp: str = None,
+        data: str = None,
+        config: str = None,
+        init: str = None,
+        logs: str = None,
+    ) -> None:
+        super().__init__()
+        self.infra = infra
+        self.var = var
+        self.cache = cache
+        self.tmp = tmp
+        self.shared_tmp = shared_tmp
+        self.data = data
+        self.config = config
+        self.init = init
+        self.logs = logs
+
+    @staticmethod
+    def from_config():
+        """Returns Localstack directory paths from the config/environment variables defined by the config."""
+        return Directories(
+            infra=INSTALL_DIR_INFRA,
+            var=TMP_FOLDER,  # TODO: add variable
+            cache=TMP_FOLDER,  # TODO: add variable
+            tmp=TMP_FOLDER,  # TODO: move default value to /tmp/localstack/host
+            shared_tmp=HOST_TMP_FOLDER,  # TODO: move default value to /tmp/localstack/shared
+            data=DATA_DIR,
+            config=None,  # TODO: will be introduced once .localstack config file has been refactored
+            init=None,  # TODO: introduce environment variable
+            logs=TMP_FOLDER,  # TODO: move to /tmp/localstack/logs
+        )
+
+    @staticmethod
+    def for_container() -> "Directories":
+        """
+        /var/lib/localstack. Returns Localstack directory paths as they are defined within the container. Everything
+        shared and writable lives in /var/lib/localstack or /tmp/localstack.
+
+        :returns: Directories object
+        """
+        return Directories(
+            infra=INSTALL_DIR_INFRA,
+            var="/var/lib/localstack/libs",
+            cache="/var/lib/localstack/cache",
+            tmp=TMP_FOLDER,  # TODO: move to /tmp/localstack
+            shared_tmp=HOST_TMP_FOLDER,  # TODO: move to /var/lib/localstack/tmp
+            data=DATA_DIR,  # TODO: move to /var/lib/localstack/data
+            config="/etc/localstack/",  # TODO: will be introduced once .localstack config file has been refactored
+            logs="/var/lib/localstack/logs",
+            init="/docker-entrypoint-initaws.d",
```

Comment:
The folders which have string values here, and are not assigned with an already existing static, are the ones which are not used yet, right?
Which basically means that this PR is non-breaking.

Reply:
correct! the ones that have static values are not used anywhere. once you start using them, any user code that relies on hardcoding of paths to get logs for example would fail.